apps/iris/src/instrumentation.go

```diff
@@ -0,0 +1,211 @@
+// internal/instrumentation/instrumentation.go
+package instrumentation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"runtime"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/shirou/gopsutil/cpu"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/log/global"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/propagation"
+	otel_log "go.opentelemetry.io/otel/sdk/log"
+	otel_metric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/feature/ec2/imds"
+)
+
+var (
+	initOnce       sync.Once
+	shutdownFuncs  []func(context.Context) error
+	tracerProvider *trace.TracerProvider
+	meterProvider  *otel_metric.MeterProvider
+	loggerProvider *otel_log.LoggerProvider
+)
+
+func getAWSInstanceID() string {
+	cfg, err := config.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		fmt.Print("failed to get Config Of AWS")
+		return "Instance-Id-Not-Found"
```

Comment:
return type을 에러 type이랑 실제 값을 분리해서 두는 게 좋아보입니다. 함수 return type도 수정이 필요할 것 같고요.

```suggestion
		return nil, fmt.Errorf("Instance-Id-Not-Found")
```